### PR TITLE
[Evaluation] Restore attack technique results in per_testing_criteria_results

### PR DESCRIPTION
**Pull request by slister1001**
Re-add attack strategy tracking in _compute_per_testing_criteria() that was removed in PR #45328. The per_testing_criteria_results field now includes both risk category entries and attack strategy entries (with attack_strategy field set), which the ACA runtime depends on for complexity aggregation.

Changes:
- Restore strategy_criteria tracking dict and attack_technique property reading in _compute_per_testing_criteria()
- Add 9 unit tests for _compute_per_testing_criteria() covering risk categories, attack strategies, aggregation, and edge cases
- Add e2e assertions in test_red_team.py and test_red_team_foundry.py validating per_testing_criteria_results contains both risk category and attack strategy entries

# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

If an SDK is being regenerated based on a new API spec, a link to the pull request containing these API spec changes should be included above.

# All SDK Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md##building-and-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/evaluation/azure-ai-evaluation/tests/e2etests/test_red_team_foundry.py [hunk 1]
```diff
@@ -76,6 +76,22 @@ def simple_target(query: str) -> str:
             assert "attack_technique" in attack
             assert attack["risk_category"] == "violence"
 
+        # Validate per_testing_criteria_results contains both risk categories and attack strategies
+        if result.scan_result:
+            summary = result.scan_result.get("aoai_compatible_summary", result.scan_result)
```
Comment (BryceByDesign) [inline sdk/evaluation/azure-ai-evaluation/tests/e2etests/test_red_team_foundry.py]: Should be "AOAI_Compatible_Summary"
